cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -112,6 +112,12 @@ double gdv_fn_random_with_seed(int64_t ptr, int32_t seed, bool seed_validity) {
   return (*holder)();
 }
 
+double gdv_fn_random_with_seed64(int64_t ptr, int64_t seed, bool seed_validity) {
+  gandiva::RandomGeneratorHolder* holder =
+      reinterpret_cast<gandiva::RandomGeneratorHolder*>(ptr);
+  return (*holder)();
```

Comment:
`seed `and `seed_validity` not used?

Reply:
The seed arg is not directly used, but just to distinguish different arg type and call corresponding func accordingly. 
About how seed be passed to gandiva to use, in random_generator_holder.cc, the seed will be obtained from func node and be used to construct RandomGeneratorHolder.

Reply:
seed_validity is not used and looks useless. I will check whether we can remove it.

Reply:
Just had a check. The rand func is registered with kResultNullNever flag, which require seed_validity. If we remove seed_validity, runtime error will happen when calling rand func.